File: CS/SchedulerControlCustomZooming/DXApplication5/Services/MyServiceWrappers.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraScheduler;
using DevExpress.Services;
using System.Windows.Forms;
using DevExpress.Services.Implementation;
using DevExpress.XtraScheduler.Native;
using DevExpress.XtraScheduler.Commands;
using DevExpress.Portable.Input;

namespace DXApplication5
{
    public class MyMouseHandlerService : MouseHandlerServiceWrapper
    {
        IServiceProvider provider;
        public MyMouseHandlerService(IServiceProvider provider, IMouseHandlerService service)
            : base(service)
        {
            this.provider = provider;
        }

        private void ZoomIn(SchedulerControl control)
        {
            MyViewZoomInCommand zoomInCommand = new MyViewZoomInCommand(control);
            zoomInCommand.Execute();
        }
        private void ZoomOut(SchedulerControl control)
        {
            MyViewZoomOutCommand zoomOutCommand = new MyViewZoomOutCommand(control);
            zoomOutCommand.MaxDayCount = 7;
            zoomOutCommand.Execute();
        }
        public override void OnMouseWheel(PortableMouseEventArgs e) {
            SchedulerMouseHandler handler = ((MouseHandlerService)this.Service).Handler as SchedulerMouseHandler;
            if(handler.IsControlPressed) {
                if(provider is SchedulerControl) {
                    SchedulerControl control = (SchedulerControl)provider;
                    if(e.Delta > 0)
                        ZoomIn(control);
                    else
                        ZoomOut(control);
                }
            }
            else
                base.OnMouseWheel(e);
        }
    }
}

File: VB/SchedulerControlCustomZooming/DXApplication5/Services/MyServiceWrappers.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraScheduler
Imports DevExpress.Services
Imports System.Windows.Forms
Imports DevExpress.Services.Implementation
Imports DevExpress.XtraScheduler.Native
Imports DevExpress.XtraScheduler.Commands
Imports DevExpress.Portable.Input

Namespace DXApplication5
	Public Class MyMouseHandlerService
		Inherits MouseHandlerServiceWrapper

		Private provider As IServiceProvider
		Public Sub New(ByVal provider As IServiceProvider, ByVal service As IMouseHandlerService)
			MyBase.New(service)
			Me.provider = provider
		End Sub

		Private Sub ZoomIn(ByVal control As SchedulerControl)
			Dim zoomInCommand As New MyViewZoomInCommand(control)
			zoomInCommand.Execute()
		End Sub
		Private Sub ZoomOut(ByVal control As SchedulerControl)
			Dim zoomOutCommand As New MyViewZoomOutCommand(control)
			zoomOutCommand.MaxDayCount = 7
			zoomOutCommand.Execute()
		End Sub
		Public Overrides Sub OnMouseWheel(ByVal e As PortableMouseEventArgs)
			Dim handler As SchedulerMouseHandler = TryCast(CType(Me.Service, MouseHandlerService).Handler, SchedulerMouseHandler)
			If handler.IsControlPressed Then
				If TypeOf provider Is SchedulerControl Then
					Dim control As SchedulerControl = DirectCast(provider, SchedulerControl)
					If e.Delta > 0 Then
						ZoomIn(control)
					Else
						ZoomOut(control)
					End If
				End If
			Else
				MyBase.OnMouseWheel(e)
			End If
		End Sub
	End Class
End Namespace